plugins/org.eclipse.glsp.api/src/org/eclipse/glsp/api/di/GLSPModule.java

```diff
@@ -52,140 +46,105 @@
-      return ContextMenuItemProvider.NullImpl.class;
+   protected Class<? extends PopupModelFactory> bindPopupModelFactory() {
+      return PopupModelFactory.NullImpl.class;
+   }
+
+   protected Class<? extends ILayoutEngine> bindLayoutEngine() {
+      return ILayoutEngine.NullImpl.class;
    }
 
-   protected Class<? extends ActionHandlerProvider> bindActionHandlerProvider() {
-      return ActionHandlerProvider.NullImpl.class;
+   protected Class<? extends ModelValidator> bindModelValidator() {
+      return ModelValidator.NullImpl.class;
    }
 
-   protected Class<? extends OperationHandlerProvider> bindOperatioHandlerProvider() {
-      return OperationHandlerProvider.NullImpl.class;
+   protected Class<? extends ActionProcessor> bindActionProcessor() {
+      return ActionProcessor.NullImpl.class;
    }
 
-   protected Class<? extends ModelExpansionListener> bindModelExpansionListener() {
-      return ModelExpansionListener.NullImpl.class;
+   protected Class<? extends LabelEditValidator> bindLabelEditValidator() {
+      return LabelEditValidator.NullImpl.class;
    }
 
-   protected Class<? extends ModelFactory> bindModelFactory() {
-      return ModelFactory.NullImpl.class;
+   protected Class<? extends ServerConfiguration> bindServerConfiguration() {
+      return ServerConfiguration.NullImpl.class;
    }
 
-   protected Class<? extends ModelSelectionListener> bindModelSelectionListener() {
-      return ModelSelectionListener.NullImpl.class;
+   protected Class<? extends CommandPaletteActionProvider> bindCommandPaletteActionProvider() {
+      return CommandPaletteActionProvider.NullImpl.class;
    }
 
-   protected Class<? extends ModelElementOpenListener> bindModelElementOpenListener() {
-      return ModelElementOpenListener.NullImpl.class;
+   protected Class<? extends ContextMenuItemProvider> bindContextMenuItemProvider() {
+      return ContextMenuItemProvider.NullImpl.class;
    }
 
-   protected Class<? extends PopupModelFactory> bindPopupModelFactory() {
-      return PopupModelFactory.NullImpl.class;
+   protected Class<? extends ToolPaletteItemProvider> bindToolPaletteItemProvider() {
+      return ToolPaletteItemProvider.NullImpl.class;
    }
 
-   protected Class<? extends ILayoutEngine> bindLayoutEngine() {
-      return ILayoutEngine.NullImpl.class;
+   protected Class<? extends ActionSupplier> bindActionSupplier() {
+      return ActionSupplier.NullImpl.class;
    }
 
-   protected Class<? extends ServerCommandHandlerProvider> bindServerCommandHandlerProvider() {
-      return ServerCommandHandlerProvider.NullImpl.class;
+   protected Class<? extends ActionHandlerSupplier> bindActionHandlerSupplier() {
+      return ActionHandlerSupplier.NullImpl.class;
    }
 
-   protected Class<? extends ModelValidator> bindModelValidator() {
-      return ModelValidator.NullImpl.class;
+   protected Class<? extends OperationHandlerSupplier> bindOperatioHandlerSupplier() {
```

Comment:
What is the difference between provider and supplier?

Reply:
Suppliers: contribute data for the server that is required for proper initialization (e.g. availableActions, Handlers etc.)
Providers: contribute optional data that is requested by (and sent to) the client (e.g. CommandPaletteActions)

Previously "Provider" was used for both these concepts which was a bit confusing and also caused weird naming schemes if you'd have a server provider of providers (e.g. RequestContextActionsProviderProvider)

But I'm open for a better approach to handle this problem :wink:

Reply:
Ok, thanks for the clarification. I see the issue now... I'll think about it and we can discuss this in the next architecture committee meeting today?